code/script.js

```diff
@@ -301,25 +462,57 @@ const filterCharacters = (keep) => {
   */
 
   // Invoke a function to redraw the board with the remaining people.
-}
+  generateBoard();
+};
+
+// Ylva - when clicking guess, the player first have to confirm that they want to make a guess.
+// If the user confirms the guess, the game will start and the checkMyGuess function will be invoked.
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
+const playersGuess = confirm(`Do you want to guess ${personToConfirm}?`);
   // remember the confirm() ?
+
   // If the player wants to guess, invoke the checkMyGuess function.
-}
+if (playersGuess) {
+checkMyGuess(personToConfirm);
+  }
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
   // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
   // 4. Hide the game board
-}
+  if (personToCheck === secret.name) {
+    // winningSound.play(); - might add sound here if time
+    winOrLoseText.innerText = `That's correct ${playerName}! The person is indeed ${secret.name}`;
+    secretImageAtCheck.innerHTML += `<img src=${secret.img} alt=${secret.name}>`;
+  } else {
+    // losingSound.play(); - might add a sound here if time
+    winOrLoseText.innerText = `You lost ${playerName}! It is not ${personToCheck}. The person is ${secret.name}`;
+    // winOrLoseSection.innerHTML += `
+    //     <img src=${secret.img} alt=${secret.name}>
+    // `;
+    secretImageAtCheck.innerHTML += `<img src=${secret.img} alt=${secret.name}>`;
+  }
+  winOrLoseSection.style.display = "flex";
+};
 
+const numberOfQuestionsLimit = () => {
+  winOrLoseSection.style.display = "flex";
+  //losingSound.play();
+  winOrLoseText.innerText = `You lost! It is not ${personToCheck}. The person is ${secret.name}`;
+};
 // Invokes the start function when website is loaded
-start()
+start();
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+restartButton.addEventListener("click", start);
+//restartButton.addEventListener('click', start),(()=>{ location.reload()}) //trying to fix game to start again
+//restartButton.addEventListener("click", start()=>{ location.reload()});
+questions.addEventListener("change", selectQuestion);
+findOutButton.addEventListener("click", checkQuestion);
+playAgainButton.addEventListener("click", start);
+findOutButton.addEventListener("click", playerNameSave);
```

Comment:
I think you put "findOutButton" here by mistake - you only want that button to call checkQuestion and nothing else

Reply:
@edv-rd 
Yeah I probably added a lot by mistake haha...
I'm sorry you had to review my piece of code as it is not working...
And you're totally right! I should have started with the basics and then stretched. But I was so happy when I got the timer to work that I thought I'd "easily" fix the rest - but I didn't...

Think your comments will help me though. And now I'm kind of re-doing the whole thing anyway so not even sure this is still the same code as the version published on Netlify haha... But that is a new challenge for tomorrow :) 
Thanks for believing in me even when I don't do that myself sometimes!

For example - this weeks Weather App feels much easier than the Guess Who game... Funny though?!